### ISSUE TITLE
Rename Date/User Columns

### ISSUE THREAD

**Issue by benhaynes** [directus/api#336]
For consistency, let's update the following and add a note in the API Contributor Docs with `*_on` (datetime) and `*_by` (user) naming conventions.

* `directus_activity.datetime` -> `directus_activity.action_on` (**thoughts on this?**)
* `directus_activity.user` -> `directus_activity.action_by` (**thoughts on this?**)
* `directus_activity.datetime_edited` -> `directus_activity.edited_on`
* `directus_activity.deleted_comment` -> `directus_activity.comment_deleted_on`

* `directus_activity_seen.seen` -> `directus_activity_seen.seen_on`

* `directus_files.upload_user` -> `directus_files.uploaded_by`
* `directus_files.upload_date` -> `directus_files.uploaded_on`

* `directus_users.last_access` -> `directus_users.last_access_on`
* `directus_users.last_login` -> `directus_users.last_login_on`

@rijkvanzanten @WellingGuzman – thoughts?

**Comment by WellingGuzman** [comment]
I prefer `*_at` better for datetime, Ex: `uploaded_at` rather than `uploaded_on`.

> `directus_activity.user` -> `directus_activity.action_by` (**thoughts on this?**)

In this case I still prefer `user`, over `action_by`. Maybe `created_by` and `created_at` rather than `action_on` and `action_by`


> `directus_activity.deleted_comment` -> `directus_activity.comment_deleted_on`
> `directus_activity_seen.seen` -> `directus_activity_seen.seen_on`

These are boolean fields, and not a date time, unless you are suggesting changing their type as well.

> `directus_users.last_access` -> `directus_users.last_access_on`

Even though the new name makes total sense, I believe `last_access` means "when".

> `directus_users.last_login` -> `directus_users.last_login_on`

Is this a new field?

**Comment by benhaynes** [comment]
I'd like to use `*_on` (not `*_at`). Since "at" is

**Comment by rijkvanzanten** [comment]
Why is this closed @benhaynes?

**Comment by benhaynes** [comment]
I'd like to use `*_on` (not` *_at`). Since "at" is ambiguous (could be location or time):
https://stackoverflow.com/questions/1379543/database-name-convention-datetime-column

> In this case I still prefer user, over action_by. Maybe created_by and created_at rather than action_on and action_by

We could use `created_*` ... @rijkvanzanten – any thoughts on this?

> These are boolean fields, and not a date time, unless you are suggesting changing their type as well.

These are datetimes on my local (from the demo schema)... how did this get updated inconsistently? Either way, it is more informative to have both of these be datetimes (not booleans). So let's plan on updating the API.

> Even though the new name makes total sense, I believe last_access means "when".

Yes, "last access" does imply "when"... but I want to be consistent and have ALL datetime fields use the same convention.

> Is this a new field?

It's in my Demo schema... Rijk?? Let's get this schema in order!

**Comment by rijkvanzanten** [comment]
> These are boolean fields, and not a date time, unless you are suggesting changing their type as well.

Didn't we decide to make it a datetime like a month ago? That way we can check value vs `null` for seen or not seen and we have the time it was seen too

**Comment by benhaynes** [comment]
Either way, let's use these ticket to update those datatypes to **DATETIME**:

* `directus_activity.deleted_comment` -> `directus_activity.comment_deleted_on`
* `directus_activity_seen.seen` -> `directus_activity_seen.seen_on`

**Comment by WellingGuzman** [comment]
> I'd like to use *_on (not*_at). Since "at" is ambiguous (could be location or time):
https://stackoverflow.com/questions/1379543/database-name-convention-datetime-column

It's true, but I always suggest using `_at` it's a more "standard" (popular from my point of view).

> Didn't we decide to make it a datetime like a month ago? That way we can check value vs null for seen or not seen and we have the time it was seen too

We were discussing this, but we never agreed on making the change.

**Comment by benhaynes** [comment]
Not sure which way is more popular, but let's update these to `*_on` to avoid ambiguity and let's convert those booleans to datetimes.

**Comment by WellingGuzman** [comment]
@benhaynes is the list on the original post is the final decision?

**Comment by benhaynes** [comment]
Yup! And these two should be `datetime`:

`directus_activity.comment_deleted_on`
`directus_activity_seen.seen_on`

**Comment by WellingGuzman** [comment]
@benhaynes On `directus_activity_seen` there's a column named `user`, Is this column stays with the same name?

**Comment by benhaynes** [comment]
Yeah, let's keep those the same. `directus_activity_seen.seen_by` seems messy.

**Comment by WellingGuzman** [comment]
@benhaynes Understood. One last thing there's a `directus_users.last_login` column, Is this going to be a new column or not? I want to make sure we keep the migrations schema up to date.

**Comment by benhaynes** [comment]
I know it used to exist... did we get rid of it? @rijkvanzanten this is in the demo schema, so it must have existed before?

It might be a nice thing to have, but if it's been removed we don't need to waste time adding it back in I guess. More important things to do.

**Comment by WellingGuzman** [comment]
It existed, but it was removed here: https://github.com/directus/api/commit/66bd4d1a2f449205df529020b1063505b0371700#diff-e99b100a76435995bdc9407845bedfc1

**Comment by WellingGuzman** [comment]
Something to notice is that the column was removed but it has a record in `directus_fields`.

Ref: https://github.com/directus/api/blob/master/migrations/db/seeds/FieldsSeeder.php#L618-L621

**Comment by rijkvanzanten** [comment]
According to our spreadsheet of columns, it has been removed, so I think it's safe to say that it's gone for good this time

**Comment by benhaynes** [comment]
OK, we can leave them out then. 

@rijkvanzanten let's plan on doing a fresh install on for the demo schemas to avoid this confusion.
